app/db/connection_manager.py

```
@@ -0,0 +1,443 @@
+                
+            except Exception as e:
+                self._connection_stats["failed_connections"] += 1
+                self._connection_stats["last_error"] = str(e)
+                logger.error(f"Failed to initialize connection pool (attempt {attempt + 1}): {e}")
+                
+                if attempt < self.config.retry_attempts - 1:
+                    wait_time = self.config.retry_delay * (2 ** attempt)  # Exponential backoff
+                    logger.info(f"Retrying in {wait_time} seconds...")
+                    time.sleep(wait_time)
+                else:
+                    raise Exception(f"Failed to initialize database connection pool after {self.config.retry_attempts} attempts")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)